monitoring/nethealth.go

```diff
@@ -328,14 +294,18 @@ func nethealthFailureProbe(name, peer string, packetLoss float64) *pb.Probe {
 	}
 }
 
-// fetchNethealthMetrics collects the network metrics from the nethealth pod
-// specified by addr. Returns the resp as an array of bytes.
-func fetchNethealthMetrics(ctx context.Context, addr string) ([]byte, error) {
-	client, err := roundtrip.NewClient(addr, "")
-	if err != nil {
-		return nil, trace.Wrap(err, "failed to connect to nethealth service at %s.", addr)
+// fetchNethealthMetrics collects the network metrics from the nethealth pod.
+// Returns the response as an array of bytes.
+func (c *nethealthChecker) fetchNethealthMetrics(ctx context.Context) (res []byte, err error) {
+	client := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				var d net.Dialer
+				return d.DialContext(ctx, "unix", c.NethealthSocketPath)
```

Comment:
nit: I'd use the passed in context instead of capturing the external one although in this case it is the same thing.